pkg/querier/queryrange/results_cache.go

```diff
@@ -202,6 +223,34 @@ func (s resultsCache) handleMiss(ctx context.Context, r Request) (Response, []Ex
 		return response, []Extent{}, nil
 	}
 
+	cacheGenNumber := s.extractor.ExtractCacheGenNumber(response)
+
+	if cacheGenNumber == "-1" {
```

Comment:
When will this be true in `handleMiss`? From what I can see this is only going to have downstream requests which will never return `-1` as the gen number?